Prog/se-vacant-lots-arlington.Rmd

```diff
@@ -0,0 +1,582 @@
+                                           "Townhouse",
+                                           "Semi-detached",
+                                           "Duplex",
+                                           "Other Residential",
+                                           "One Family Dwelling")),
+         one_unit_flag = ifelse(dwelling_type == "One Family Dwelling",
+                                1,
+                                0),
+         two_unit_flag = ifelse(dwelling_type == "Duplex",
+                                1,
+                                0))
+
+
+```
+
+## Merge with parcel data
+ 
+Now that we have selected the correct records from the zoning code file, we can merge with the cleaned property records data. First, we have to do a bit of cleaning of these codes, including getting rid of instances where there is a tie, and correcting some of the notation. We will create a new variable in order to archive the original `zoning` variable.
+
+
+```{r bk-clean}
+
+jurfinal <- jur %>% 
+  mutate(zoning_code = case_when(
+    zoning == "RC" ~ "R-C",
+    zoning == "C-0 Crystal City" ~ "C-O Crystal City",
+    zoning %in% c("C-O-ROSSLYN", "C-O-ROSS") ~ "C-O Rosslyn",
+    zoning == "RA-H-3.2" ~ "RA-H 3.2",
+    zoning == "C-1-O" ~ "C-1-0",
+    TRUE ~ zoning
+  ))
+           
+```
+
+Merge on zoning code variable. Test that all variables merged.
+
+```{r zoning-merge}
+
+zoningdata <- left_join(jurfinal, zoningcode2, by = "zoning_code")
+
+```
+
+There are `r nrow(anti_join(jurfinal, zoningcode1, by = c("zoning_code")))` observations that did not have a match in the zoning code. Below, we see which zoning codes were in the Black Knight data but NOT in the zoning code data.
+
+```{r antijoin}
+
+anti_join(jurfinal, zoningcode1, by = c("zoning_code")) %>% 
+  count(zoning) %>% 
+  knitr::kable()
+
+```
+
+Aside from those with missing zoning information, these zoning codes do not allow for residential properties, which explains why they were not found in the zoning code data.
+
+### Break out special exceptions
+
+Two datasets will be created: one that includes special exceptions, and one that doesn't. This way, we do not throw out any observations that have no special exceptions, or any lots that have the appropriate lot size for the by-right dwelling type, but not the special exception.
+
+There are many lots that are too small to build on, but we still need to include them. Create a flag to identify them. We can first group by both the address and the flag, and take the most dense dwelling type. This way, we can won't filter out observations that are too small, but we will can use the lot size to eliminate less-dense dwelling types, if they exist. We can then group by the address again and take the minumum of the `smallsize_flag`, so we take the dwelling type that is appropriate for the lot size, if it exists.
+
+
+```{r filter-small}
+
+byright <- zoningdata %>% 
```

Comment:
so we lose about 2000 obs between jurfinal and specialexception/byright - this is expected?

Reply:
yes, this is a combination of parcels that are not zoned for residential and parcels with missing zoning